extensions/control-plane/provision/provision-bigquery/src/main/java/org/eclipse/edc/connector/provision/gcp/BigQueryProvisioner.java

```diff
@@ -93,14 +91,11 @@ public CompletableFuture<StatusResult<ProvisionResponse>> provision(
             }
             monitor.info("BigQuery Provisioner table " + target.getTableName().toString() + " exists");
 
-            GcpAccessToken token = null;
-
-            if (serviceAccount != null) {
-                token = iamService.createAccessToken(serviceAccount);
-            } else {
-                serviceAccount = ADC_SERVICE_ACCOUNT;
-                token = iamService.createDefaultAccessToken();
+            if (serviceAccount == null) {
```

Comment:
this would be more readable as `else` branch of the if at line 82:
```java
            GcpServiceAccount serviceAccount;
            if (serviceAccountName != null) {
                serviceAccount = iamService.getServiceAccount(serviceAccountName);
            }  else {
                serviceAccount = IamService.ADC_SERVICE_ACCOUNT;
            }
```

or, as a oneliner:
```java
var serviceAccount = Optional.ofNullable(serviceAccountName)
                    .map(iamService::getServiceAccount)
                    .orElse(IamService.ADC_SERVICE_ACCOUNT);
```
 
In fact, this logic is not needed at all, because the iamService already does that internally!
```java
var serviceAccount = iamService.getServiceAccount(serviceAccountName);
```
is all we need. 
Same thing in the `GcsProvisioner` class.

Reply:
The code can be simplified, definitely! I would still delegate the iamService object to decide what serviceAccount is appropriate. Working on it..

Reply:
Done